rlclientlib/logger/async_batcher.h

```diff
@@ -46,27 +49,30 @@ namespace reinforcement_learning { namespace logger {
 
     int run_iteration(api_status* status) override;
 
+	void flush() override; //flush all batches
+
   private:
     int fill_buffer(std::shared_ptr<utility::data_buffer>& retbuffer,
       size_t& remaining, 
       api_status* status);
 
-    void flush(); //flush all batches
 
   public:
     async_batcher(i_message_sender* sender,
                   utility::watchdog& watchdog,
                   shared_state_t& shared_state,
                   error_callback_fn* perror_cb,
+                  i_trace *tracer,
```

Comment:
nit: should we place the `trace` argument to the end since it is more likely to be missing than `config`?

Reply:
I find optional arguments in non-API methods to be a terribly bad idea as it makes REALLY EFFING HARD to correctly wire them around.

It's not a compiler error to miss an optional argument and that will lead to stuff mysteriously not working.
There's no ergonomic value in optional argument for non-published APIs.

I made the change this way so i_trace doesn't have to be optional.